vmdk_plugin/vmdkops/esx_vmdkcmd.go

```diff
@@ -85,19 +86,30 @@ func (vmdkCmd EsxVmdkCmd) Run(cmd string, name string, opts map[string]string) (
 	defer C.free(unsafe.Pointer(ans))
 
 	// connect, send command, get reply, disconnect - all in one shot
-	_, err = C.Vmci_GetReply(C.int(EsxPort), cmdS, beS, ans)
-
-	if err != nil {
-		var errno syscall.Errno
-		errno = err.(syscall.Errno)
-		msg := fmt.Sprintf("'%s' failed: %v (errno=%d).", cmd, err, int(errno))
-		if errno == syscall.ECONNRESET {
-			msg += " Hit communication issue with ESX (vmci or ESX service)\n"
-			msg += " Please refer to the FAQ https://github.com/vmware/docker-volume-vsphere/wiki#faq"
+	retryCount := 10
```

Comment:
Yes, these ops aren't idempotent and a retry will fail while a disk may have been created or worse attached and the client doesn't know about it.

Instead perk these failures upwards and return a "transient" error to the caller and the caller can decide whether to retry. The retry must be done by first fetching current status of the volume (say on attach)  or figure if the volume is there (create/delete) and only then decide whether a retry is needed.

For this issue can may be generate the error and a separate issue to do the error handling appropriately.